pkg/collector/corechecks/system/disk/diskv2/disk_windows.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+
+package diskv2
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	win "golang.org/x/sys/windows"
+	"regexp"
+	"strings"
+	"unsafe"
+)
+
+func compileRegExp(expr string) (*regexp.Regexp, error) {
+	iExpr := fmt.Sprintf("(?i)%s", expr)
+	return regexp.Compile(iExpr)
+}
+
+func (c *Check) fetchAllDeviceLabelsFromLsblk() error {
+	return nil
+}
+
+func (c *Check) fetchAllDeviceLabelsFromBlkidCache() error {
+	return nil
+}
+
+func (c *Check) fetchAllDeviceLabelsFromBlkid() error {
+	return nil
+}
+
+var mpr = win.NewLazySystemDLL("mpr.dll")
+var procWNetAddConnection2W = mpr.NewProc("WNetAddConnection2W")
+
+type netResource struct {
+	Scope       uint32
+	Type        uint32
+	DisplayType uint32
+	Usage       uint32
+	localName   *uint16
+	remoteName  *uint16
+	comment     *uint16
+	provider    *uint16
+}
+
+// RemotePath constructs the remote path based on the mount type.
+// It converts the Type to uppercase to ensure case-insensitive evaluation.
+func (m mount) RemotePath() (string, string) {
+	if strings.TrimSpace(m.Type) == "" {
+		m.Type = "SMB"
+	}
+	// Convert Type to uppercase for case-insensitive comparison
+	normalizedType := strings.ToUpper(strings.TrimSpace(m.Type))
+	if normalizedType == "NFS" {
+		return normalizedType, fmt.Sprintf(`%s:%s`, m.Host, m.Share)
+	}
+	var userAndPassword string
+	if len(m.User) > 0 {
+		userAndPassword += m.User
+	}
+	if len(m.Password) > 0 {
+		userAndPassword += fmt.Sprintf(":%s", m.Password)
+	}
+	if len(userAndPassword) > 0 {
+		return normalizedType, fmt.Sprintf(`\\%s@%s\%s`, userAndPassword, m.Host, m.Share)
+	}
+	return normalizedType, fmt.Sprintf(`\\%s\%s`, m.Host, m.Share)
```

Comment:
I don't see this logic in the Python code, is that expected ? (eg. is it hidden behind the windows syscalls ?)

The code I see would just be something like 
```suggestion
	return normalizedType, strings.TrimRight(fmt.Sprintf("\\%s\%s", m.Host, m.Share), `\`)
```